src/login/logind-dbus.c

```diff
@@ -3256,6 +3115,15 @@ static int method_set_wall_message(
         if (r < 0)
                 return r;
 
+        /* sysvinit has a 252 (256-(strlen(" \r\n")+1)) character
+         * limit for the wall message. There is no real technical
+         * need for that but doesn't make sense to store arbitrary
+         * armounts either.
+         * https://git.savannah.nongnu.org/cgit/sysvinit.git/tree/src/shutdown.c#n72)
+        */
+        if (strlen(wall_message) > 252)
```

Comment:
given how arbitrary this number is, please add a macro for it.

Please return a proper dbus error i.e.

```c
return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Wall message too long, maximum permitted length is %u characters.", WALL_MESSAGE_MAX);
```